File: mycollab-dao/src/main/java/com/esofthead/mycollab/core/arguments/BooleanSearchField.java
/**
 * This file is part of mycollab-dao.
 *
 * mycollab-dao is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-dao is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-dao.  If not, see <http://www.gnu.org/licenses/>.
 */
/**
 * This file is part of mycollab-core.
 *
 * mycollab-core is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-core is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-core.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.core.arguments;

/**
 * 
 * @author MyCollab Ltd.
 * @since 1.0
 */
public class BooleanSearchField extends SearchField {
	private static final long serialVersionUID = 1L;

	public static final String IS = "is";

	private boolean value;

	private String comparision;

	public BooleanSearchField() {
		this(false);
	}

	public BooleanSearchField(boolean value) {
		this(SearchField.AND, value);
	}

	public BooleanSearchField(String oper, boolean value) {
		this(oper, BooleanSearchField.IS, value);
	}

	public BooleanSearchField(String oper, String comparision, boolean value) {
		this.operation = oper;
		this.comparision = comparision;
		this.value = value;
	}

	public boolean isValue() {
		return value;
	}

	public void setValue(boolean value) {
		this.value = value;
	}

	public String getComparision() {
		return comparision;
	}

	public void setComparision(String comparision) {
		this.comparision = comparision;
	}
}

File: mycollab-dao/src/main/java/com/mycollab/db/arguments/BooleanSearchField.kt
/**
 * Copyright © MyCollab
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.db.arguments

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
class BooleanSearchField(oper: String, var comparision: String?, var isValue: Boolean) : SearchField() {

    @JvmOverloads constructor(value: Boolean = false) : this(SearchField.AND, value)

    constructor(oper: String, value: Boolean) : this(oper, BooleanSearchField.IS, value)

    init {
        this.operation = oper
    }

    companion object {
        private val serialVersionUID = 1L

        val IS = "is"
    }
}
